packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -113,20 +124,65 @@ function setScopedDecoratorData(
     return;
   }
 
-  // if scope specified, create or overwrite with the new value
-  const splitScopes = scope.split(",").map((s) => s.trim());
-  const targetEntry = context.program.stateMap(key).get(target);
-
-  // if target doesn't exist in decorator map, create a new entry
-  if (!targetEntry) {
-    const newObject = Object.fromEntries(splitScopes.map((scope) => [scope, value]));
+  const [negationScopes, scopes] = parseScopes(context, scope);
+  if (negationScopes !== undefined && negationScopes.length > 0) {
+    const newObject = Object.fromEntries([AllScopes].map((scope) => [scope, value]));
+    newObject[negationScopesKey] = negationScopes;
     context.program.stateMap(key).set(target, newObject);
     return;
+  } else if (scopes !== undefined && scopes.length > 0) {
+    // if scope specified, create or overwrite with the new value
+    const targetEntry = context.program.stateMap(key).get(target);
+
+    // if target doesn't exist in decorator map, create a new entry
+    if (!targetEntry) {
+      const newObject = Object.fromEntries(scopes.map((scope) => [scope, value]));
```

Comment:
nit, since we always need the newObject(whether targetEntry is empty or not), we could consider put it outside, and do `if(!targetEntry)` later. This way we could replace the `return` with `else`.

Reply:
Right. It was like this in the first place, simplified.